packages/stg-evm-v2/test/devtools/utils.test.ts

```diff
@@ -94,4 +100,117 @@ describe('devtools/utils', () => {
+            ])
+        })
+
+        it('should throw an error for invalid chain names', () => {
+            const chains = [
+                'ethereum-mainnet',
+                'invalid-chain-1',
+                'arbitrum-mainnet',
+                'invalid-chain-2',
+                'base-mainnet',
+            ]
+            expect(() => getContracts(chains, mockContractData, isValidCreditMessagingChain)).to.throw(
+                'Invalid chains found: invalid-chain-1, invalid-chain-2'
+            )
+        })
+
+        it('should trim whitespace from chain names', () => {
+            const chains = [' ethereum-mainnet', 'arbitrum-mainnet ', '  base-mainnet']
+            const result = getContracts(chains, mockContractData, isValidCreditMessagingChain)
+            expect(result.length).to.equal(3)
+            expect(result.map((r) => r.eid)).to.have.members([
+                EndpointId.ETHEREUM_V2_MAINNET,
+                EndpointId.ARBITRUM_V2_MAINNET,
+                EndpointId.BASE_V2_MAINNET,
+            ])
+        })
+
+        it('should return all valid contracts for an empty input array', () => {
+            const result = getContracts([], mockContractData, isValidCreditMessagingChain)
+            expect(result.length).to.equal(validCreditMessagingChains.size)
+        })
+    })
+
+    describe('filterConnections', () => {
+        const mockConnections = [
+            { from: { eid: 1 }, to: { eid: 2 }, data: 'A' },
+            { from: { eid: 2 }, to: { eid: 3 }, data: 'B' },
+            { from: { eid: 3 }, to: { eid: 1 }, data: 'C' },
+            { from: { eid: 1 }, to: { eid: 3 }, data: 'D' },
+            { from: { eid: 2 }, to: { eid: 1 }, data: 'E' },
+            { from: { eid: 4 }, to: { eid: 5 }, data: 'F' },
+        ]
+
+        it('should filter connections based on from and to contracts', () => {
+            const mockFromContracts = [{ eid: 1 }, { eid: 2 }]
+            const mockToContracts = [{ eid: 2 }, { eid: 3 }]
+            const result = filterConnections(mockConnections, mockFromContracts, mockToContracts)
+            expect(result.length).to.equal(3)
+            expect(result).to.deep.include({ from: { eid: 1 }, to: { eid: 2 }, data: 'A' })
+            expect(result).to.deep.include({ from: { eid: 1 }, to: { eid: 3 }, data: 'D' })
+            expect(result).to.deep.include({ from: { eid: 2 }, to: { eid: 3 }, data: 'B' })
+        })
+
+        it('should handle overlapping from and to contracts', () => {
+            const mockFromContracts = [{ eid: 1 }, { eid: 2 }, { eid: 3 }]
+            const mockToContracts = [{ eid: 1 }, { eid: 2 }, { eid: 3 }]
+            const result = filterConnections(mockConnections, mockFromContracts, mockToContracts)
+            expect(result.length).to.equal(5)
+            expect(result).to.not.deep.include({ from: { eid: 4 }, to: { eid: 5 }, data: 'F' })
+        })
+
+        it('should return empty array when no connections match', () => {
```

Comment:
slightly confused here by the test description `should return empty array` when the return is an array of length 1